bin/oneoff/remove_non_teacher_application_ids.rb

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+
+puts "Removing application ids that are not teacher applications ..."
+modified_enrollments = []
+
+Pd::Enrollment.where.not(application_id: nil).each do |enrollment|
+  original_app_id = enrollment.application_id
+
+  begin
+    application = Pd::Application::ApplicationBase.find(original_app_id)
+    if application.type != "Pd::Application::TeacherApplication"
+      # The method set_application_id is triggered before a save
+      # Save the enrollment to re-compute set_application_id
+      enrollment.save!
+      modified_enrollments << enrollment.id
+    end
+  rescue ActiveRecord::SubclassNotFound # Happens with Facilitator applications
+    enrollment.save!
```

Comment:
I had to do a double take on this too, so it's maybe worth an extra comment.

`application = Pd::Application::ApplicationBase.find(original_app_id)` will throw `ActiveRecord::SubclassNotFound` error if it's a facilitator application. So the associated enrollment will save only as part of the rescue block.

But `application = Pd::Application::ApplicationBase.find(original_app_id)` may *not* throw an error if it's a principal approval application that got linked. So then it saves only on the begin block, not the rescue block.

Reply:
Perhaps the rescue block should happen earlier to make it clear that it's failing on the `find` method? Open to recs on how to make this clearer

Reply:
ahh, I see! I agree it would be clearer to rescue only the `find` command. however, since this is a oneoff and you've already tested it on the adhoc, I feel good about this level of clarification without you having to rerun it.

Reply:
Sounds good! Thanks